packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/amplify-meta-updaters.ts

```diff
@@ -106,10 +106,24 @@ export const getPostUpdateAuthMetaUpdater = (context: any) => async (resourceNam
   return resourceName;
 };
 
-function getFrontendConfig(authParameters: AuthParameters) {
+export function getFrontendConfig(authParameters: AuthParameters) {
   const loginMechanisms: string[] = [];
   loginMechanisms.push(...(authParameters?.aliasAttributes || []).map((att: string) => att.toUpperCase()));
 
+  // backwards compatibility
+  if (authParameters?.usernameAttributes) {
+    const usernameAttributes = authParameters?.usernameAttributes[0];
+    if (usernameAttributes.includes(',')) {
+      usernameAttributes
+        .toUpperCase()
+        .replace(/\s/g, '')
+        .split(',')
+        .forEach(it => loginMechanisms.push(it));
+    } else {
+      loginMechanisms.push(usernameAttributes.toUpperCase());
+    }
+  }
+
```

Comment:
Do we need to do a check for `if (loginMechanisms.length === 0)` and add the username as a default?

See: https://github.com/aws-amplify/amplify-cli/pull/8093#issuecomment-912630619